### DISCUSSION TITLE
What is the best approach to build a website on top of Directus CMS?

### DISCUSSION THREAD

**Discussion by kropsi** [directus/directus#11476] [Archive]
I was looking to find the best option for a static website generator or another way of building a website on top of Directus CMS. Interestingly, there is no unambiguous answer to this problem. Did someone find an approach that works well and goes well hand in hand with Directus?

**Comment by luciandavidescu** [comment]
Most often, this would be done with a frontend such as Gatsby or Next.js or others like those.
Or you can just use a templating engine such as handlebars or ejs etc. and write your own logic to wrap them.

You might want to look into the code I linked here: https://github.com/luciandavidescu/directus-static-frontend/blob/main/directus-static-frontend.js / explained from here: https://github.com/directus/directus/discussions/10981

(relies on webhooks to build pages at generation time)

While not a mainstream approach and not working out of the box, it might help give an idea of where to start.

**Comment by rashidpathiyil** [comment]
Try Nuxt.js.

**Reply by casualmatt** [reply]
sadly there is no official nuxt3 support from Directus.

**Reply by rijkvanzanten** [reply]
> sadly there is no official nuxt3 support from Directus.

What do you mean by no official support? There's nothing stopping you from using Nuxt 3 with Directus 🙂

**Reply by casualmatt** [reply]
hello @rijkvanzanten, i mean that there is not [official plugin for Nuxt](https://modules.nuxtjs.org/?version=3.x), and ofc you can do it manually but is not the same experience that what you can have with other backend that have an official plugin + documentation (for nuxt).

**Reply by rijkvanzanten** [reply]
True, just keep in mind that the official "plugin" would be nothing more than a wrapper around `@directus/sdk`, so the actual developer experience is going to be 98% identical between just using the SDK or using a Nuxt 3 package that we built 🙂

**Reply by genu** [reply]
For visibility, to wrap the directus sdk in a nuxt plugin, you can just do this:

you just need to define the env var and the `AppData` type if using typescript 

````ts 
import { Directus } from '@directus/sdk'
import { AppData } from './collections/app-data'

export default defineNuxtPlugin(() => {
  const config = useRuntimeConfig()

  return {
    provide: {
      directus: new Directus<AppData>(config.public.baseApi),
    },
  }
})
````

Then you can use:

````ts 
const { $directus } = useNuxtApp()
````

Anywhere in your app

**Reply by Loosie94** [reply]
This can be a good starter: https://github.com/bryantgillespie/nuxt3-directus-starter :)
